librz/analysis/arch/arm/arm_il32.c

```diff
@@ -2314,22 +2515,880 @@ static RzILOpEffect *tbb(cs_insn *insn, bool is_thumb) {
+				break;
+			case ARM_INS_VCLE:
+				cond = INV(FORDER(r_elem, l_elem));
+				break;
+			case ARM_INS_VCLT:
+				cond = FORDER(l_elem, r_elem);
+				break;
+			case ARM_INS_VACGE:
+				cond = INV(FORDER(FABS(l_elem), FABS(r_elem)));
+				break;
+			case ARM_INS_VACGT:
+				cond = FORDER(FABS(r_elem), FABS(l_elem));
+				break;
+			default:
+				cond = NULL;
+				rz_il_op_pure_free(l_elem);
+				rz_il_op_pure_free(r_elem);
+				rz_il_op_effect_free(eff);
+				return NULL;
+			}
+
+			eff = SEQ2(eff,
+				write_reg_lane(REGID(0), i, vec_size,
+					ITE(cond, LOGNOT(UN(vec_size, 0)), UN(vec_size, 0))));
+		}
+		return eff;
+	}
+
+	// for integer number
+	ut32 vec_size = DT_WIDTH(insn);
+	RzILOpEffect *eff = NULL;
+	for (int i = 0; i < REG_WIDTH(0) / vec_size; ++i) {
+		RzILOpBitVector *l_elem = read_reg_lane(REGID(1), i, vec_size);
+		RzILOpBitVector *r_elem = ISIMM(2) ? UN(vec_size, 0) : read_reg_lane(REGID(2), i, vec_size);
+
+		RzILOpBool *cond;
+		bool as_signed = is_vec_signed(VVEC_DT(insn));
+		switch (insn->id) {
+		case ARM_INS_VCEQ:
+			cond = EQ(l_elem, r_elem);
+			break;
+		case ARM_INS_VCGE:
+			cond = as_signed ? SGE(l_elem, r_elem) : UGE(l_elem, r_elem);
+			break;
+		case ARM_INS_VCGT:
+			cond = as_signed ? SGT(l_elem, r_elem) : UGT(l_elem, r_elem);
+			break;
+		case ARM_INS_VCLE:
+			cond = as_signed ? SLE(l_elem, r_elem) : ULE(l_elem, r_elem);
+			break;
+		case ARM_INS_VCLT:
+			cond = as_signed ? SLT(l_elem, r_elem) : SLE(l_elem, r_elem);
+			break;
+		default:
+			cond = NULL;
+			rz_il_op_pure_free(l_elem);
+			rz_il_op_pure_free(r_elem);
+			rz_il_op_effect_free(eff);
+			return NULL;
+		}
+
+		eff = SEQ2(eff,
+			write_reg_lane(REGID(0), i, vec_size,
+				ITE(cond, LOGNOT(UN(vec_size, 0)), UN(vec_size, 0))));
+	}
+
+	return eff;
+}
+
+static RzILOpEffect *vtst(cs_insn *insn, bool is_thumb) {
+	if (OPCOUNT() < 3) {
+		return NULL;
+	}
+
+	// vtst <Vd>, <Vn>, <Vm>
+	// for each lane:
+	// Vd = iszero((n and m)) ? zero : not(zero)
+	ut32 vec_size = VVEC_SIZE(insn);
+	RzILOpEffect *eff = NULL;
+	for (int i = 0; i < REG_WIDTH(0) / vec_size; ++i) {
+		RzILOpBitVector *n = read_reg_lane(REGID(1), i, vec_size);
+		RzILOpBitVector *m = read_reg_lane(REGID(2), i, vec_size);
+		RzILOpBitVector *d = ITE(IS_ZERO(LOGAND(n, m)),
+			UN(vec_size, 0),
+			LOGNOT(UN(vec_size, 0)));
+		eff = SEQ2(eff,
+			write_reg_lane(REGID(0), i, vec_size, d));
+	}
+
+	return eff;
+}
+
+static RzILOpEffect *vldn_multiple_elem(cs_insn *insn, bool is_thumb) {
+	ut32 rm_idx = OPCOUNT() - 1;
+	ut32 rn_idx;
+	ut32 regs = 0;
+	bool wback = insn->detail->arm.writeback;
+	bool use_rm_as_wback_offset = false;
+	ut32 n = insn->id - ARM_INS_VLD1 + 1;
+
+	// vldn {list}, [Rn], Rm
+	if (!ISMEM(rm_idx)) {
+		regs = OPCOUNT() - 2;
+		use_rm_as_wback_offset = true;
+	} else {
+		// vldn {list}, [Rn]
+		rm_idx = -1;
+		regs = OPCOUNT() - 1;
+	}
+
+	// mem_idx
+	rn_idx = regs;
+
+	// assert list_size % n == 0
+	// assert they were all Dn
```

Comment:
Those asserts can be checked with `rz_warn_if...` functions. They get removed in the Release build